coalib/misc/CachingUtilities.py

```diff
@@ -0,0 +1,174 @@
+import calendar
+import os
+import pickle
+import time
+
+from coalib.output.Tagging import get_user_data_dir
+
+
+def get_cache_data_path(log_printer, filename):
+    """
+    Get the full path of ``filename`` present in the user's cache directory.
+
+    :param log_printer: A LogPrinter object to use for logging.
+    :param filename:    The file whose path needs to be expanded.
+    :return:            Full path of the file, assuming it's present in the
+                        user's config directory.
+    """
+    return os.path.join(get_user_data_dir(
+        log_printer, action="caching"), filename)
+
+
+def delete_cache_files(log_printer, files):
+    """
+    Delete the cache files after displaying a warning saying the cache
+    is corrupted and will be removed.
+
+    :param log_printer: A LogPrinter object to use for logging.
+    :param files:       The list of files to be deleted.
+    :return:            True if all the given files were successfully deleted.
+                        False otherwise.
+    """
+    error_files = []
+    for file_name in files:
+        file_path = get_cache_data_path(log_printer, file_name)
+        cache_dir = os.path.dirname(file_path)
+        try:
+            os.remove(file_path)
+        except OSError:
+            error_files.append(file_name)
+
+    if len(error_files) > 0:
+        error_files = ", ".join(error_files)
+        log_printer.warn("There was a problem deleting the following "
+                         "files: " + error_files + ". Please delete "
+                         "them manually from " + cache_dir)
```

Comment:
I don't think `error_files` needs quotes though since it's MD5 sums and won't have spaces. But agreed on `cache_dir`.